### PR TITLE
Create dinic.py

### PR DESCRIPTION
**Pull request by danielx285**
Dinic's algorithm for maximum flow

### CODE REVIEW

File: graphs/dinic.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        self.lvl = [0] * n
+        self.ptr = [0] * n
+        self.q = [0] * n
+        self.adj = [[] for _ in range(n)]
+
+    def add_edge(self, a, b, c, rcap=0):
+        self.adj[a].append([b, len(self.adj[b]), c, 0])
+        self.adj[b].append([a, len(self.adj[a]) - 1, rcap, 0])
+
+    def dfs(self, v, t, f):
```
Comment (cclauss) [inline graphs/dinic.py]: Please make API understandable to the first time user.  In of `dfs(self, v, t, f)`, something more like `determine_fastest_surfer(self, viscosity, timber, finances)`as discussed in CONTRIBUTING.md.  Also, these methods/functions need doctests.

File: graphs/dinic.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python2.7
```
Comment (cclauss) [inline graphs/dinic.py]: Please remove line 1.  This repo no longer supports legacy Python.

File: graphs/dinic.py (hunk 1 above)
Comment (danielx285) [reply graphs/dinic.py]: Ok, I will change the file.

**Comment by cclauss** [comment]
https://docs.python.org/3/library/doctest.html is what is needed on key functions/methods.

Can we have a better name for `rcap` ?

**Pull request merged**